### PR TITLE
Visualization of Sieve of Eratosthenes algorithm

### PR DESCRIPTION
**Pull request by hs2361**
This is to address issue #180 
This is a visualization of the Sieve of Eratosthenes algorithm for finding prime numbers below any natural number 'n'.

### CODE REVIEW

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import pygame
```
Comment (prateekiiest) [inline Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: Add pygame dependency

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            clock.tick()
+            count += 1
+        else:
+            break
+pygame.display.update()
+# Sieving algorithm
+# all the numbers except 1 since 1 is neither prime nor composite
+nums = set(range(2, num + 1))
+composites = {1}  # set of composites detected so far
+celldict[1].highlight()  # highlight the Cell 1
```
Comment (prateekiiest) [inline Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: highlight is a method for class Cell not celldict.
Explain more here

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    pygame.event.pump()
+    for j in range(0, width, 32):
+        if (count <= num):
+            # create a Cell object at every grid intersection
+            cell = Cell(j, i, count)
+            clock.tick()
+            count += 1
+        else:
+            break
+pygame.display.update()
```
Comment (prateekiiest) [inline Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: why update method here?

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        # highlight the primes with GREEN
+        pygame.draw.rect(display, GREEN, (x, y, 32, 32))
+        message_display(f"{val}", x + 16, y + 16, 17)
+        pygame.time.wait(100)  # to create a moving highlight
+        primetxt += str(val) + ' ,'  # concatenate primes to primetxt string
+
+# Displaying results using GUI
+m = tkinter.Tk()  # root window
+l1 = tkinter.Label(m, text='Primes that were sieved: ')
+# Message widget to display the sieved primes
```
Comment (prateekiiest) [inline Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: Rearrange code in a more well-fashioned way.
First have all class definitions and methods and normal functions.

Then have the main code [those without any indent]

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py (hunk 2 above)
Comment (hs2361) [reply Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: Celldict[1] refers to the Cell having value 1. So, we are calling the .highlight() method on the Cell object, and not on the celldict

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py (hunk 1 above)
Comment (hs2361) [reply Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: pygame in the requirements.txt file?

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py (hunk 3 above)
Comment (hs2361) [reply Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: The update() method was called here in order to show the Cells on the pygame window. Without it, the Cells would not have appeared until the next call of update()

File: Code-Sleep-Python/Prime/sieve_of_eratosthenes.py (hunk 2 above)
Comment (hs2361) [reply Code-Sleep-Python/Prime/sieve_of_eratosthenes.py]: Celldict[1] refers to the Cell having value 1. So, we are calling the .highlight() method on the Cell object, and not on the celldict

**Comment by prateekiiest** [comment]
make the changes in this PR only, by pushing more commits here

**Comment by prateekiiest** [comment]
Please post an output of the code here

**Comment by hs2361** [comment]
1) image

2) The program actually produces an animation of highlighting all the cells, but here I am only uploading a screenshot. 
image

3) image

**Comment by prateekiiest** [comment]
Looks good!
Thanks for the contribution @hs2361  🎉 🎉 .
Feel free to add yourself under the contributor list

**Pull request merged**